### PR TITLE
Removed afterburner usage from Cosmos Exception

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Removed afterburner usage from Cosmos Exception
* Closes https://github.com/Azure/azure-sdk-for-java/issues/19847
* Closes https://github.com/Azure/azure-sdk-for-java/issues/20088
* On investigating this more, the issue started happening on `CosmosException` after we converted the cosmos exception message into Json format in this PR : https://github.com/Azure/azure-sdk-for-java/pull/18092 after release 4.11.0 and above. 
* Currently just logging `CosmosDiagnostics` doesn't throw this warning, but when logging `CosmosException`, the warning is thrown, because `CosmosException` uses `Utils.getSimpleObjectMapper()` - which registers module `AfterBurnerModule()`
```
Utils.simpleObjectMapper.registerModule(new AfterburnerModule());
```
There is no need to use this same ObjectMapper for `CosmosException`, it can use its own `ObjectMapper` as used by `CosmosDiagnostics`, `ClientSideRequestStatistics`, `ClientTelemetry`, `RntbdObjectMapper` and so on.

### CODE REVIEW

**Review by moderakh — APPROVED** [review]
Thanks @kushagraThapar 
LGTM.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 public class CosmosException extends AzureException {
     private static final long serialVersionUID = 1L;
 
-    private static final ObjectMapper mapper = Utils.getSimpleObjectMapper();
+    private static final ObjectMapper mapper = new ObjectMapper();
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java:47]: BTW, request diagnostics also use an ObjectMapper. don't we need to do similiar fix to that?

**Review by FabianMeiswinkel — APPROVED** [review]
As discussed offline - I would prefer also renaming Utils.getSimpleObjectMapper to something like Utils.getObjectMapperWithAfterBurner to avoid unintentional regressions in the future.

Otherwise LGTM

**Comment by moderakh** [comment]
> As discussed offline - I would prefer also renaming Utils.getSimpleObjectMapper to something like Utils.getObjectMapperWithAfterBurner to avoid unintentional regressions in the future.
> 
> Otherwise LGTM

can we maybe call it `Utils.getObjectMapperForPayloadData()` or something like that as it is intended to be used for serializing/deserializing data payload and afterburner is intended to optimize that code path only.

**Review by xinlian12 — APPROVED** [review]
LGMT, thanks for the fix Kushagra

**Comment by kushagraThapar** [comment]
@FabianMeiswinkel @moderakh  - I have updated the name to `getObjectMapperForPayLoadData`. 

The reason I didn't choose `objectMapperWithAfterBurner` is because in near future if we register more modules to this objectmapper, then having afterBurner in the name might not justify the different modules this objectmapper refers to. 
Let me know if this makes sense or not.

**Pull request merged**